wasp/src/main/java/com/orhanobut/wasp/WaspImageHandler.java

```diff
@@ -0,0 +1,177 @@
+package com.orhanobut.wasp;
+
+import android.graphics.Bitmap;
+import android.os.Looper;
+import android.text.TextUtils;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import com.orhanobut.wasp.utils.StringUtils;
+
+/**
+ * This class is responsible of the loading image. It automatically handles the canceling and
+ * loading images for the recycled view as well.
+ *
+ * @author Orhan Obut
+ */
+final class WaspImageHandler implements ImageHandler {
+
+    /**
+     * It is used to determine which url is current for the ImageView
+     */
+    private static final int KEY_TAG = 0x7f070006;
+
+    /**
+     * Stores the cached images
+     */
+    private final ImageCache imageCache;
+
+    /**
+     * It is used to create network request for the bitmap
+     */
+    private final ImageNetworkHandler imageNetworkHandler;
+
+    WaspImageHandler(ImageCache cache, ImageNetworkHandler handler) {
+        this.imageCache = cache;
+        this.imageNetworkHandler = handler;
+    }
+
+    @Override
+    public void load(final WaspImage waspImage) {
+        checkMain();
+        loadImage(waspImage);
+    }
+
+    private void loadImage(final WaspImage waspImage) {
+        final String url = waspImage.getUrl();
+        final ImageView imageView = waspImage.getImageView();
+
+        // clear the target
+        initImageView(waspImage);
+
+        // update the current url
+        imageView.setTag(KEY_TAG, url);
+
+        int width = imageView.getWidth();
+        int height = imageView.getHeight();
+
+        boolean wrapWidth = false;
+        boolean wrapHeight = false;
+        if (imageView.getLayoutParams() != null) {
+            wrapWidth = imageView.getLayoutParams().width == ViewGroup.LayoutParams.WRAP_CONTENT;
+            wrapHeight = imageView.getLayoutParams().height == ViewGroup.LayoutParams.WRAP_CONTENT;
+        }
+
+        // if the view's bounds aren't known yet, and this is not a wrap-content/wrap-content
+        // view, hold off on loading the image.
+        boolean isFullyWrapContent = wrapWidth && wrapHeight;
+        if (width == 0 && height == 0 && !isFullyWrapContent) {
+            Logger.d("ImageHandler : width == 0 && height == 0 && !isFullyWrapContent");
+            // return;
+        }
+
+        // Calculate the max image width / height to use while ignoring WRAP_CONTENT dimens.
+        int maxWidth = wrapWidth ? 0 : width;
+        int maxHeight = wrapHeight ? 0 : height;
+
+        // if there is any old request. cancel it
+        String tag = (String) imageView.getTag(KEY_TAG);
+        if (tag != null) {
```

Comment:
At line 53, a tag is set to ImageView which makes this if always true. Therefor, this check should be moved to the beginning of method.

By the way, previous request may be with the same url. So, I think we should also check if tag is not equal with current url.